src/coreComponents/constitutive/solid/CoupledSolid.hpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+
+/**
+ * @file CoupledSolid.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_SOLID_COUPLEDSOLID_HPP_
+#define GEOSX_CONSTITUTIVE_SOLID_COUPLEDSOLID_HPP_
+
+#include "constitutive/ConstitutiveBase.hpp"
+#include "constitutive/solid/SolidBase.hpp"
+
+namespace geosx
+{
+namespace constitutive
+{
+
+template< typename SOLID_TYPE,
+          typename PORO_TYPE >
+class CoupledSolidUpdates
+{
+public:
+  /**
+   * @brief Constructor
+   */
+  CoupledSolidUpdates( SOLID_TYPE * solidModel,
+                       PORO_TYPE * porosityModel ):
+    m_solidUpdate( solidModel->createKernelUpdates() ),
+    m_porosityUpdate( porosityModel->createKernelUpdates() )
+  {}
+
+  /// Deleted default constructor
+  CoupledSolidUpdates() = delete;
+
+  /// Default copy constructor
+  CoupledSolidUpdates( CoupledSolidUpdates const & ) = default;
+
+  /// Default move constructor
+  CoupledSolidUpdates( CoupledSolidUpdates && ) = default;
+
+  /// Deleted copy assignment operator
+  CoupledSolidUpdates & operator=( CoupledSolidUpdates const & ) = delete;
+
+  /// Deleted move assignment operator
+  CoupledSolidUpdates & operator=( CoupledSolidUpdates && ) =  delete;
+
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  typename SOLID_TYPE::KernelWrapper const getSolidModel() const
+  {
+    return m_solidUpdate;
+  }
```

Comment:
no, not yet but because the `CompressibleSolid` is not used yet. I think it will be useful. I will remove it if not.

Reply:
You should maybe remove it now (or at least comment the code).
It will be easy to put it back when you need it.
Because it's really possible that you forget to remove it in the end if useless.

Reply:
sounds good.